src/views/register/RegisterView.js

```diff
@@ -49,16 +52,74 @@ class RegisterView extends React.Component {
         this.setState(obj);
         this.setState({disabled: !this.validInput()});
     }
-    
+
+    pickImage = async () => {
+        let pickerResult = await ImagePicker.launchImageLibraryAsync({
+            allowsEditing: true,
+            aspect: [4, 3],
+            quality: 0.8,
+            base64: true,
+            mediaTypes: ImagePicker.MediaTypeOptions.Images 
+        });
+        
+        if (!pickerResult.cancelled) {
+            console.log("avatarUrl:", pickerResult.uri)
+            this.setState({ avatarUrl: pickerResult.uri, disabled: false});
+            let base64Img = `data:image/jpg;base64,${pickerResult.base64}`
+            this.base64Img = base64Img;
+        }
+    };
+
+    updateAvatar = async () =>{
+        this.setState({spinner: true})
+        try {
+            //Cloudinary api
+            let apiUrlUpload = 'https://api.cloudinary.com/v1_1/startach/image/upload';
+            let userFolder = this.state.email || this.state.phone;
+            let dataUp = {
+                "file": this.base64Img,
+                "upload_preset": 'atuhnhof',
+                "public_id": userFolder + '_' + Date.now(),
+                "folder": 'avatars/' + userFolder ,
+            }
+            
+            await fetch(apiUrlUpload, {
+            body: JSON.stringify(dataUp),
+            headers: {
+                'content-type': 'application/json'
+            },
+            method: 'POST',
+            })
+            .then( response => {
+            let data = response._bodyText
+            let newAvatarUrl = JSON.parse(data).secure_url;
+            console.log("Avatar URL:",newAvatarUrl);
+            this.setState({ avatarUrl: newAvatarUrl });
+            })
+            .catch(err => console.log(err))
+            
+        } catch (e) {
+            console.log(e);
+            alert('Upload failed',e);
+        }
+        finally{
+            this.setState({spinner: false})
+        }      
+    };
+
     render() {
       return(
         <ImageBackground style={styles.background} source={require('../../images/backGround.jpg')}>
         <KeyboardAvoidingView
             style={styles.topContainer}
             behavior="padding">
             <View style={styles.userView}>
-                <TouchableOpacity>
-                <Image style = {styles.userImage} source ={require('../../images/emptyUserIcon.png')} />
+                <TouchableOpacity onPress={this.pickImage}>
+                    { this.state.avatarUrl ? 
+                        <Image style = {styles.userImage} source={{ uri: this.state.avatarUrl }} />
+                        :
+                        <Image style = {styles.emptyUserImage} source = {require('../../images/emptyUserIcon.png')} />  
+                    }
```

Comment:
Sometimes it makes it easier to understand when you do it as well:
```js
const avatarImage = this.state.avatarUrl ? (
  <Image style={styles.userImage} source={{ uri: this.state.avatarUrl }} />
) : (
  <Image
    style={styles.emptyUserImage}
    source={require('../../images/emptyUserIcon.png')}
  />
);

return (
  <TouchableOpacity onPress={this.pickImage}>{avatarImage}</TouchableOpacity>
)
```

Reply:
Thanks for the review, I hope the changes are good now.
Nice to meet you